cache/cache.go

```diff
@@ -191,3 +194,23 @@ func (c *Cache) Path(checksum, uri string) string {
 	base := BasePath(checksum, uri)
 	return filepath.Join(c.root, base)
 }
+
+// UnavailableError returns 101 for the exit code.
+type UnavailableError struct {
```

Comment:
Oh you'll need to implement `Unwrap() error` in order to work with `errors.Is()`: https://pkg.go.dev/errors#Unwrap